tiny_dnn/layers/recurrent_cell_layer.h

```diff
@@ -53,15 +52,17 @@ class recurrent_cell_layer : public layer {
    * @param in_dim [in] number of elements of the input
    * @param out_dim [in] number of elements of the output
    * @param has_bias [in] whether to include additional bias to the layer
+   * @param activation [in] activation function to be used internally
    **/
   recurrent_cell_layer(serial_size_t in_dim,
                        serial_size_t out_dim,
-                       bool has_bias          = true,
-                       backend_t backend_type = core::default_engine())
+                       bool has_bias                = true,
+                       activation_layer *activation = new tanh_layer,
```

Comment:
I thought it would be easier at the user level to write rnn_cell(in, out, bias, new tanh) than rnn_cell(in, out, bias, std::shared_ptr<tanh>()) but I don't really mind chaging it for a shared pointer.